typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +145,221 @@ def touch_import(package, name, node):
 
     create_import(package, name, node)
     return name
+
+
+_block_syms = {syms.funcdef, syms.classdef, syms.trailer}
+
+
+def _find(name, node):
+    nodes = [node]
+    while nodes:
+        node = nodes.pop()
+        if node.type > 256 and node.type not in _block_syms:
+            nodes.extend(node.children)
+        elif node.type == token.NAME and node.value.strip() == name:
+            return node
+    return None
+
+
+ImportPairing = NamedTuple("ImportPairing", [
+    ("entry", str),
+    ("binding_name", str)
+])
+
+ImportNodeInfo = NamedTuple("ImportNodeInfo", [
+    ("imports", Dict[str, Set[ImportPairing]]),
+    ("node", Node)
+])
+
+ImportInfo = NamedTuple("ImportInfo", [
+    ("entry", str),
+    ("package", str),
+    ("binding", str)
+])
+
+
+def find_import_info(package, name, node):
+    # type: (Optional[str], str, Node) -> Optional[ImportInfo]
+    """
+    Finds the import statement for <name> regardless of whether <name> is the binding name
+        ex where name='a':
+        import a => match AND import a as b => match BUT import b as a => None
+
+    Parameters
+    -----------
+    package : Optional[str]
+    name : str
+    node : Node
+
+    Return
+    -----------
+    Optional[ImportInfo]
+    """
+    for child in node.children:
+        if is_import(child):
+            ret = get_import_info(child)
+            if package is None:
+                package = 'no_pkg'
```

Comment:
I'm not a fan of making up sentinel values like this, unless there's not a good alternative, and/or doing so just makes our lives much easier.

In this case, I think using an empty string would work just as well (and we wouldn't have to worry about a package that actually has this name.)  We could also just use `None`, as that's a hashable value which can be used as a key in dictionaries... but that would require changing the type of the dict.

So... if we're change from `None` to `''` - the next question is, why don't we just use `''` instead of `None` to begin with - ie, make the type of `package` be just `str` instead of `Optional[str]`? And I actually think that would make sense, if for no reason that it would simplify the typing.  However, if you'd rather minimize changes and stick with `None`, that's fine too.